### ISSUE TITLE
 [main] ERROR ws.schild.jave.FFMPEGExecutor - Could not find ffmpeg executable for native/ffmpeg-amd64.exe is the correct platform jar included?

### ISSUE THREAD

**Issue by CodingOX** [a-schild/jave2#35]
I user maven import jar like this in win10 and jdk8:
```
<dependency>
            <groupId>ws.schild</groupId>
            <artifactId>jave-all-deps</artifactId>
            <version>2.4.4</version>
 </dependency>
```
and use code like this
```
public class MediaTimeCount {
    public static void main(String[] args) throws EncoderException {
        System.out.println("getInfo");
        File file = new File("I:\\xxx.avi");
        MultimediaObject instance = new MultimediaObject(file);
        MultimediaInfo result = instance.getInfo();
        long duration = result.getDuration();
        System.out.println(duration);
    }
}
```
but console print error message like this:
```
21:41:24.110 [main] DEBUG ws.schild.jave.FFMPEGExecutor - Executable path: C:\Users\Victory\AppData\Local\Temp\jave\ffmpeg-amd64-2.4.4.exe
Exception in thread "main" java.lang.NullPointerException
	at java.util.Objects.requireNonNull(Objects.java:203)
	at java.nio.file.Files.copy(Files.java:2984)
	at ws.schild.jave.DefaultFFMPEGLocator.copy(DefaultFFMPEGLocator.java:144)
	at ws.schild.jave.DefaultFFMPEGLocator.copyFile(DefaultFFMPEGLocator.java:123)
	at ws.schild.jave.DefaultFFMPEGLocator.<init>(DefaultFFMPEGLocator.java:84)
	at ws.schild.jave.MultimediaObject.<init>(MultimediaObject.java:60)
	at club.hicode.media.MediaTimeCount.main(MediaTimeCount.java:20)
21:41:24.124 [main] ERROR ws.schild.jave.FFMPEGExecutor - Could not find ffmpeg executable for native/ffmpeg-amd64.exe is the correct platform jar included?
```
should i download ffmpeg-amd64.exe and copy it to C:\Users\Victory\AppData\Local\Temp\jave ??

**Comment by a-schild** [comment]
For me this works, how do you run the project?
You need all the referenced jar's in your classpath, as show in the deps-view of the pom
![2018-11-15 19_18_05-apache netbeans ide 9 0](https://user-images.githubusercontent.com/1441009/48572997-51bc3100-e90b-11e8-9505-0444a0eab4f3.png)

**Comment by WavyPeng** [comment]
Should I extend `FFMPEGLocator` and override `getFFMPEGExecutablePath()` method?

**Comment by a-schild** [comment]
Can you please check, if the jave-native-win32-2.4.5.jar is in your project?
I apparently looks for the 32-bit executable

**Comment by WavyPeng** [comment]
> Can you please check, if the jave-native-win32-2.4.5.jar is in your project?
> I apparently looks for the 32-bit executable

Sure I am. I include it in my maven project, as follows:
![image](https://user-images.githubusercontent.com/26155781/50077772-49b02380-0220-11e9-95a9-64c707373e25.png)
Is there anything wrong with my configuration? I really don't know how to solve it. Thank you :)

**Comment by a-schild** [comment]
When you run the project, do you see the jar file in the lib folder of your project?

**Comment by WavyPeng** [comment]
> When you run the project, do you see the jar file in the lib folder of your project?

I include the `jave-native-win32-2.4.5.jar` file in the lib folder, as follows:

![image](https://user-images.githubusercontent.com/26155781/50131173-ed024680-02bb-11e9-87cf-1bf73823c001.png)

However, it still has the same problem. I found when I ran my project, a `jave` folder would be created in path `C:\Users\..\AppData\Local\Temp`, but there was no content in it. 

![image](https://user-images.githubusercontent.com/26155781/50131373-cc86bc00-02bc-11e9-8a59-585c93e9d56e.png)

 Should I copy the `jave-native-win32-2.4.5.jar` into this folder? I'm not sure about the loading path for `ffmpeg` when using `jave`. Thank you :)

**Comment by WavyPeng** [comment]
> When you run the project, do you see the jar file in the lib folder of your project?

Thanks very much for the help! `Jave2` is a very nice project and is useful for me 👍 .

I debugged my project and finally found where I did wrong. 
The archtechture of my OS is `x86`，so the source file should be `ffmpeg-x86.exe` instead of the file `ffmpeg-i386.exe` provided by you. That's why my project couldn't find the source file and copy it into path `C:\Users\..\AppData\Local\Temp\jave`. 

![image](https://user-images.githubusercontent.com/26155781/50144742-e0e1ad80-02ea-11e9-8077-32c7e3943a3d.png)

![image](https://user-images.githubusercontent.com/26155781/50144791-fc4cb880-02ea-11e9-8e69-fef49522ae3d.png)

![image](https://user-images.githubusercontent.com/26155781/50144880-31590b00-02eb-11e9-9ba9-6bbac26f8453.png)

Therefore, I rename the file `ffmpeg-i386.exe` to `ffmpeg-x86.exe`, and everything goes well.

**Comment by a-schild** [comment]
Thanks for tracking this down.
Can you try with the 2.4.6-SNAPSHOT version?
Shoudl fix this issue

**Comment by WavyPeng** [comment]
Ok, I will. Thanks for your help :)

**Comment by a-schild** [comment]
Should be fixed with the 2.4.6 final release

**Comment by runnermann** [comment]
Hi, This still seems to be an issue. I installed using Gradle. I am using osx and Java 1.8 . I resolved the issue by manually installing the ffmpeg-x86_64osx into the appropriate folder provided in the error message. I also had to rename the file by adding the "-2.6.0-osx" at the end of the file name so it matched the error. This resolved this problem.
